provider/openrouter/openrouter.go

```diff
@@ -51,14 +54,37 @@ func (p *Provider) ID() (id string) {
 	return "openrouter"
 }
 
+// ModelsList holds a list of models.
+type ModelsList struct {
+	Models []Model `json:"data"`
+}
+
+// Model holds a model.
+type Model struct {
+	// ID holds the model id.
+	ID string `json:"id"`
+	// Pricing holds the pricing information of a model.
+	Pricing Pricing `json:"pricing"`
+}
+
+// Pricing holds the pricing information of a model.
+type Pricing struct {
+	// Prompt holds the price for a prompt.
+	Prompt string `json:"prompt"`
+	// Completion holds the price for a completion.
+	Completion string `json:"completion"`
+	// Request holds the price for a request.
+	Request string `json:"request"`
+	// Image holds the price for a image.
+	Image string `json:"image"`
+}
+
 // Models returns which models are available to be queried via this provider.
 func (p *Provider) Models() (models []model.Model, err error) {
-	client := p.client()
-
-	var responseModels openai.ModelsList
+	var responseModels ModelsList
```

Comment:
I think it is time we add some tests for this. Can you just call the function (should work without an API token) and make sure there is no error and the list of models is not empty? Should suffice for now :angel: